autoPyTorch/api/base_task.py

```diff
@@ -173,7 +173,7 @@ def __init__(
         self._dataset_requirements: Optional[List[FitRequirement]] = None
         self._metric: Optional[autoPyTorchMetric] = None
         self._logger: Optional[PicklableClientLogger] = None
-        self.run_history: Optional[RunHistory] = None
+        self.run_history: Dict = {}
```

Comment:
Do you think it is possible to make this a:
```
Optional[Runhistory] = None
```
There is rich information in the runhistory, like the configuration to id, among other things. If we only store the dictionary, we lose them.